src/xrc/src/api.rs

```diff
@@ -235,17 +236,76 @@ fn validate(rate: QueriedExchangeRate) -> Result<QueriedExchangeRate, ExchangeRa
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+enum NormalizedTimestampType {
+    /// The timestamp is within the past minute.
+    RequestedOrCurrent,
+    /// The timestamp is within 1-2 minutes ago.
+    Past,
+}
+
+#[derive(Debug)]
+struct NormalizedTimestamp {
+    /// The timestamp in seconds.
+    value: u64,
+    /// Used to determine if the timestamp is a current timestamp or a timestamp from a minute ago.
+    r#type: NormalizedTimestampType,
+}
+
+impl NormalizedTimestamp {
+    /// Creates a new timestamp with the type `RequestedOrCurrent`
+    fn requested_or_current(value: u64) -> Self {
+        Self {
+            value,
+            r#type: NormalizedTimestampType::RequestedOrCurrent,
+        }
+    }
+
+    /// Creates a new timestamp with the type `Past`
+    fn past(value: u64) -> Self {
+        Self {
+            value,
+            r#type: NormalizedTimestampType::Past,
+        }
+    }
+}
+
+/// If the request contains a timestamp, the function returns the normalized requested timestamp.
+/// If the request's timestamp is null, the function returns the current timestamp if no assets are found
+/// to be inflight, OR the normalized timestamp from one minute ago.
+fn get_normalized_timestamp(
+    env: &impl Environment,
+    request: &GetExchangeRateRequest,
+) -> NormalizedTimestamp {
+    if request.timestamp.is_some() {
+        let timestamp = utils::get_normalized_timestamp(env, request);
+        return NormalizedTimestamp::requested_or_current(timestamp);
+    }
+
+    let timestamp = utils::get_normalized_timestamp(env, request);
+    let base_asset_is_inflight = matches!(request.base_asset.class, AssetClass::Cryptocurrency)
```

Comment:
Why do you use `matches!` here?
Why is this check even needed? Only cryptocurrencies can be in-flight, right? So, if the second check returns `true`, the first must be `true`.

Reply:
Good point. Removed.